lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -687,6 +707,35 @@ instance FromJSON (ApiT (WalletDelegation (ApiT PoolId))) where
 instance ToJSON (ApiT (WalletDelegation (ApiT PoolId))) where
     toJSON = genericToJSON walletDelegationOptions . getApiT
 
+instance FromJSON ApiWalletDelegation where
+    parseJSON = genericParseJSON defaultRecordTypeOptions
+instance ToJSON ApiWalletDelegation where
+    toJSON = genericToJSON defaultRecordTypeOptions
+
+instance FromJSON ApiWalletDelegationNext where
+    parseJSON = withObject "apiWalletDelegationNext" $ \o -> do
+        t  <- o .:? "target"
```

Comment:
[`.:?`](http://hackage.haskell.org/package/aeson-1.4.6.0/docs/Data-Aeson.html#v:.:-63-) returns `Nothing` if the key is not there or if the value is `null`.

Here you're determining the delegation status based off the existence of the `target` field.

It would seem more logical to me to first parse and case split on the `delegation_status`.

If the status is `delegating`, then the `target` field _must_ exist, right?

Reply:
yeah, we have either :

Reply:
but still we can be more rigorous in case and check ALSO if status is delegating when target is present

Reply:
done in https://github.com/input-output-hk/cardano-wallet/pull/1318/commits/d4aca5887172e02285be2e9582cbae5cecc15887

Reply:
I think your use of `.:?` still would accept `{"status": "not_delegating", "target": null}` as valid.

Is this desirable?

Also, are the `Aeson.String` patterns necessary? Ambiguous string type otherwise?

Reply:
I treat `Aeson.String "delegating/not_delegating" as additional check here. If other things will appear we make it `fail`. I will merge it as it is and also think about `null` stuff (probably will add examples to show the case) in next PR concluding this U/S